### PR TITLE
Use navigitor.userAgent to determine isMobile

### PR DESCRIPTION
**Pull request by shall0pass**
This is an alternate method to determine the isMobile status for using the responsive mobile view.  This should respect the "Desktop site" option if used in a mobile browser.

Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent

Reference: https://www.tutorialspoint.com/javascript-how-to-detect-if-a-website-is-open-on-a-mobile-or-a-desktop

### CODE REVIEW

**Comment by tjfinlinson** [comment]
I did some testing of this method and it seems to recognize a desktop vs mobile browser. I'm testing on Android and Windows. So it might be nice for someone to test this on Desktop Safari and Mobile Safari on ios devices.

On the desktop Brave Browser it did not go to the responsive app version when shrinking the window.
On the mobile brave browser it opened the responsive app version first and then checking desktop version in setting switched the the desktop view
On the mobile brave browser when unchecking desktop view it switches back to the responsive version.

This alternative method to determine isMobile seems to work as intended. If it works in ios it might be a good method to implement.

**Comment by trevdor** [comment]
Hey all! Sorry, I’ve been traveling for work.

Thanks, @shall0pass, for jumping on this so fast after the topic arose about supporting Request Desktop Site. 

I have more philosophical thoughts to explain perhaps tomorrow when I have a minute, but a practical question first:

Is there a tidy way to stop requesting the desktop site if someone does it but decides they liked the mobile view later? And/or, does “Request Desktop Site” create a long-term or temporary preference on most platforms? (Haven’t had a chance to try yet, myself.)

**Comment by shall0pass** [comment]
In my experience, the "Desktop site" option in the browser is a temporary preference for that session (I don't know if that's universal).  It will be cleared when closing the browser tab.  On Android, the preference isn't saved when creating a shortcut either, so the app would open in mobile as default and only show the desktop portion by toggling the setting.  To go back to mobile in the same session, I just uncheck the setting.

**Comment by shall0pass** [comment]
The more I read, the more I see recommendations against this approach alone because it is easily broken.  Some suggest combining the UserAgent string with feature detection and screen width/screen height.  There is a quite long regex string on [detectobilebrowsers.com](http://detectmobilebrowsers.com) to read the UserAgent string for mobile devices.

I've seen [modernizr](https://modernizr.com/docs) pitched multiple times.  It has a function to detect hidpi screens, since pixel width isn't necessarily the best indicator by itself.  Detecting touch isn't so useful anymore since a lot of laptops come with touch screens.  I'm not certain what other features would be desktop or mobile specific.  Side thought: This tool can also detect IndexDB and IndexDB blob compatibility.  May be useful in other areas of the app?

Another idea would be to use the UserAgent to initially set a screen but have an in-app preference that would remember the preferred state and could override the initial state.  Devils advocate: This isn't functionally different than the "Desktop Site" option found in browsers, with the exception of a stored preference.

Complaint: This is way more complicated than it should be.  It's too bad there isn't a more widely implemented method to do this by now.

**Comment by j-f1** [comment]
IMO we should keep the check simple for now and expand it later if people complain. And ideally over time most components will be updated to be a single component that smoothly adjusts its layout as window size changes.

**Comment by trevdor** [comment]
How about making this an either/or for starters? Under 600px OR the user-agent match? I agree, user agent sniffing has a bad name for a reason, but I think it meets our temporary needs here and as an OR it won't need to be bulletproof.

tl;dr - I'm good with shipping some version of this PR, but also want us to pivot away from `isMobile` asap

Agree with @j-f1's last comment--I'd love for us to get away from the `isMobile` concept altogether. When you scale your window down on desktop, a responsive site should still be usable. We could offer a preference so someone can lock into any layout for cases where they prefer something other than the default layout on their device. 

Our current stopgap approach is kinda the 10-years-ago thing of `mobile.website.com` serving the "mobile optimized" version with limited functionality. Essentially a whole different app that duplicates features from the desktop version and has to be kept from diverging in weird ways. 

I would like for most all our features to work at narrow widths. (At least, every feature we have so far seems adaptable to small screens with some design effort.) You could snap an Actual window to one side of your screen and a bank statement window to the other, and always have a useful Actual.

Lastly, I started experimenting with putting the available size is on a top-level context (@j-f1 or @shall0pass might have been the one to suggest that in the first place) and could easily be overridden from a preference. But that's not ready and this PR is.

**Comment by j-f1** [comment]
If we use a width check then users under 600px won't be able to access the desktop version at all, even if they request desktop site.

**Comment by trevdor** [comment]
> If we use a width check then users under 600px won't be able to access the desktop version at all, even if they request desktop site.

Good point. I think this is why a preference is nice when your device doesn't match (in any way) the requirements for the layout you prefer. However, "Request Desktop Site" should only be a temporary workaround, imo. That's made for that pre-responsive site that segregated users based on device sniffing. (Again, I'm cool with temporarily leaning on this to give folks an immediate fix--we're already explicitly serving two layouts based on `isMobile`, so no philosophical change here--but imo RDS should not be part of our long-term solution.)

**Comment by j-f1** [comment]
> but imo RDS should not be part of our long-term solution

strong agree.

**Comment by trevdor** [comment]
I've been distracted lately, but I haven't seen anyone try this branch and say it didn't yield the mobile/desktop site at the right time. Should we merge it as-is? 
(I'll leave it to you @j-f1.s)

**Review by j-f1 — APPROVED** [review]
Let’s give this a try!
